commit 439316380574b6dc8b3c3ed37efa1e74656c342f
Author: Gilles Caulier <caulier.gilles@gmail.com>
Date:   Tue May 28 14:01:58 2013 +0200

    factoring everywhere Genreal config group used to store main windows settings in RC file
    Ensure that full screen settings is loaded at initialization of AlbumGUI.
    CCBUGS: 320371
    CCBUGS: 320016
    CCBUGS: 319876

diff --git a/digikam/main/digikamapp.cpp b/digikam/main/digikamapp.cpp
index 9e38514..782e64c 100644
--- a/digikam/main/digikamapp.cpp
+++ b/digikam/main/digikamapp.cpp
@@ -190,7 +190,9 @@ DigikamApp::DigikamApp()
 
     setObjectName("Digikam");
 
-    if (d->config->group("General Settings").readEntry("Show Splash", true) &&
+    KConfigGroup group = AlbumSettings::instance()->generalConfigGroup();
+
+    if (group.readEntry("Show Splash", true) &&
         !kapp->isSessionRestored())
     {
         d->splashScreen = new SplashScreen();
@@ -208,7 +210,7 @@ DigikamApp::DigikamApp()
             + QString::number(QCoreApplication::instance()->applicationPid()));
 
     // collection scan
-    if (d->config->group("General Settings").readEntry("Scan At Start", true) ||
+    if (group.readEntry("Scan At Start", true) ||
         !CollectionScanner::databaseInitialScanDone())
     {
         ScanController::instance()->completeCollectionScanDeferFiles(d->splashScreen);
@@ -220,7 +222,6 @@ DigikamApp::DigikamApp()
     }
 
     // ensure creation
-    AlbumSettings::instance();
     AlbumManager::instance();
     LoadingCacheInterface::initialize();
     IccSettings::instance()->loadAllProfilesProperties();
@@ -266,7 +267,7 @@ DigikamApp::DigikamApp()
     // This manager must be created after collection setup.
     d->tagsActionManager = new TagsActionMngr(this);
 
-    applyMainWindowSettings(d->config->group("General Settings"));
+    applyMainWindowSettings(group);
 
     // Check ICC profiles repository availability
 
@@ -291,10 +292,11 @@ DigikamApp::DigikamApp()
     // preload additional windows
     preloadWindows();
 
-    setAutoSaveSettings("General Settings", true);
+    readFullScreenSettings(group);
+    setAutoSaveSettings(group, true);
 
     // Now, enable finished the collection scan as deferred process
-    NewItemsFinder* tool = new NewItemsFinder(NewItemsFinder::ScanDeferredFiles);
+    NewItemsFinder* const tool = new NewItemsFinder(NewItemsFinder::ScanDeferredFiles);
     tool->start();
 
     LoadSaveThread::setInfoProvider(new DatabaseLoadSaveFileInfoProvider);
@@ -2414,7 +2416,7 @@ void DigikamApp::slotSetupChanged()
     }
 
     // Load full-screen options
-    KConfigGroup group = AlbumSettings::instance()->defaultConfigGroup();
+    KConfigGroup group = AlbumSettings::instance()->generalConfigGroup();
     readFullScreenSettings(group);
 
     d->view->applySettings();
@@ -2446,7 +2448,7 @@ void DigikamApp::slotEditKeys()
 
 void DigikamApp::slotConfToolbars()
 {
-    saveMainWindowSettings(d->config->group("General Settings"));
+    saveMainWindowSettings(AlbumSettings::instance()->generalConfigGroup());
     KEditToolBar dlg(factory(), this);
 
     connect(&dlg, SIGNAL(newToolBarConfig()),
@@ -2457,7 +2459,7 @@ void DigikamApp::slotConfToolbars()
 
 void DigikamApp::slotNewToolbarConfig()
 {
-    applyMainWindowSettings(d->config->group("General Settings"));
+    applyMainWindowSettings(AlbumSettings::instance()->generalConfigGroup());
 }
 
 void DigikamApp::slotConfNotifications()
diff --git a/digikam/utils/albumsettings.cpp b/digikam/utils/albumsettings.cpp
index a7db340..6cda14d 100644
--- a/digikam/utils/albumsettings.cpp
+++ b/digikam/utils/albumsettings.cpp
@@ -552,9 +552,9 @@ void AlbumSettings::init()
             this, SLOT(applyNepomukSettings()));
 }
 
-KConfigGroup AlbumSettings::defaultConfigGroup() const
+KConfigGroup AlbumSettings::generalConfigGroup() const
 {
-    return d->config->group(d->configGroupDefault);
+    return d->config->group(d->configGroupGeneral);
 }
 
 void AlbumSettings::readSettings()
@@ -563,7 +563,7 @@ void AlbumSettings::readSettings()
 
     // ---------------------------------------------------------------------
 
-    KConfigGroup group  = defaultConfigGroup();
+    KConfigGroup group = config->group(d->configGroupDefault);
 
     QStringList collectionList = group.readEntry(d->configAlbumCollectionsEntry, QStringList());
 
@@ -655,7 +655,7 @@ void AlbumSettings::readSettings()
 
     // ---------------------------------------------------------------------
 
-    group = config->group(d->configGroupGeneral);
+    group = generalConfigGroup();
 
     d->showSplash                = group.readEntry(d->configShowSplashEntry,                      true);
     d->useTrash                  = group.readEntry(d->configUseTrashEntry,                        true);
@@ -701,7 +701,7 @@ void AlbumSettings::saveSettings()
 
     // ---------------------------------------------------------------------
 
-    KConfigGroup group = defaultConfigGroup();
+    KConfigGroup group = config->group(d->configGroupDefault);
 
     group.writeEntry(d->configAlbumCollectionsEntry,             d->albumCategoryNames);
     group.writeEntry(d->configAlbumSortOrderEntry,               (int)d->albumSortOrder);
@@ -775,7 +775,7 @@ void AlbumSettings::saveSettings()
 
     // ---------------------------------------------------------------------
 
-    group = config->group(d->configGroupGeneral);
+    group = generalConfigGroup();
 
     group.writeEntry(d->configShowSplashEntry,                   d->showSplash);
     group.writeEntry(d->configUseTrashEntry,                     d->useTrash);
diff --git a/digikam/utils/albumsettings.h b/digikam/utils/albumsettings.h
index 90197d3..e0b3816 100644
--- a/digikam/utils/albumsettings.h
+++ b/digikam/utils/albumsettings.h
@@ -393,7 +393,7 @@ public:
     void setApplicationStyle(const QString& style);
     QString getApplicationStyle() const;
 
-    KConfigGroup defaultConfigGroup() const;
+    KConfigGroup generalConfigGroup() const;
 
 public Q_SLOTS:
 
diff --git a/utilities/setup/setupalbumview.cpp b/utilities/setup/setupalbumview.cpp
index 166de47..8039650 100644
--- a/utilities/setup/setupalbumview.cpp
+++ b/utilities/setup/setupalbumview.cpp
@@ -110,11 +110,11 @@ public:
 SetupAlbumView::SetupAlbumView(QWidget* const parent)
     : QScrollArea(parent), d(new Private)
 {
-    QWidget* panel      = new QWidget(viewport());
+    QWidget* const panel      = new QWidget(viewport());
     setWidget(panel);
     setWidgetResizable(true);
 
-    QVBoxLayout* layout = new QVBoxLayout(panel);
+    QVBoxLayout* const layout = new QVBoxLayout(panel);
 
     // --------------------------------------------------------
 
@@ -306,7 +306,7 @@ void SetupAlbumView::applySettings()
     settings->setShowFolderTreeViewItemsCount(d->showFolderTreeViewItemsCount->isChecked());
     settings->saveSettings();
 
-    KConfigGroup group = settings->defaultConfigGroup();
+    KConfigGroup group = settings->generalConfigGroup();
     d->fullScreenSettings->saveSettings(group);
 }
 
@@ -357,7 +357,7 @@ void SetupAlbumView::readSettings()
     d->previewShowIcons->setChecked(settings->getPreviewShowIcons());
     d->showFolderTreeViewItemsCount->setChecked(settings->getShowFolderTreeViewItemsCount());
 
-    KConfigGroup group = settings->defaultConfigGroup();
+    KConfigGroup group = settings->generalConfigGroup();
     d->fullScreenSettings->readSettings(group);
 }
 
diff --git a/libs/widgets/common/dxmlguiwindow.cpp b/libs/widgets/common/dxmlguiwindow.cpp
index ebec2dd..b577d68 100644
--- a/libs/widgets/common/dxmlguiwindow.cpp
+++ b/libs/widgets/common/dxmlguiwindow.cpp
@@ -178,7 +178,8 @@ void DXmlGuiWindow::slotToggleFullScreen(bool set)
 
         // restore sidebars
 
-        showSideBars(true);
+        if ((d->fsOptions & FS_SIDEBARS) && d->fullScreenHideSideBars)
+            showSideBars(true);
 
         // restore thummbbar
 
diff --git a/libs/widgets/common/sidebar.cpp b/libs/widgets/common/sidebar.cpp
index 4df7e92..2dfee0c 100644
--- a/libs/widgets/common/sidebar.cpp
+++ b/libs/widgets/common/sidebar.cpp
@@ -44,6 +44,7 @@
 #include <kdeversion.h>
 #include <kglobal.h>
 #include <kiconloader.h>
+#include <kdebug.h>
 
 namespace Digikam
 {
@@ -85,7 +86,9 @@ public:
 
     bool                          minimizedDefault;
     bool                          minimized;
-    bool                          isMinimized;      // Backup of minimized status (used with Fullscreen)
+    bool                          isMinimized;      // Backup of shrinked status before backup(), restored by restore()
+                                                    // NOTE: when sidebar is hidden, only icon bar is affected. If sidebar view is 
+                                                    // visible, this one must be shrink and restored accordingly.
 
     int                           tabs;
     int                           activeTab;
@@ -182,12 +185,11 @@ void Sidebar::doSaveState()
 
 void Sidebar::backup()
 {
+    // backup preview state of sidebar view (shrink or not)
     d->isMinimized = d->minimized;
-
-    if (!d->isMinimized)
-    {
-        shrink();
-    }
+    
+    // In all case, shrink sidebar view 
+    shrink();
 
     KMultiTabBar::hide();
 }
@@ -205,12 +207,13 @@ void Sidebar::backup(const QList<QWidget*> thirdWidgetsToBackup, QList<int> *siz
 
 void Sidebar::restore()
 {
+    KMultiTabBar::show();
+
+    // restore preview state of sidebar view, stored in backup()
     if (!d->isMinimized)
     {
         expand();
     }
-
-    KMultiTabBar::show();
 }
 
 void Sidebar::restore(const QList<QWidget*> thirdWidgetsToRestore, const QList<int>& sizes)
diff --git a/digikam/main/digikamapp.cpp b/digikam/main/digikamapp.cpp
index 782e64c..c28014e 100644
--- a/digikam/main/digikamapp.cpp
+++ b/digikam/main/digikamapp.cpp
@@ -3028,7 +3028,7 @@ void DigikamApp::showThumbBar(bool visible)
 
 bool DigikamApp::thumbbarVisibility() const
 {
-    return d->view->isThumbBarVisible();
+    return d->showBarAction->isChecked();
 }
 
 }  // namespace Digikam
diff --git a/digikam/views/digikamview.cpp b/digikam/views/digikamview.cpp
index f7242a0..475bfdf 100644
--- a/digikam/views/digikamview.cpp
+++ b/digikam/views/digikamview.cpp
@@ -1952,11 +1952,6 @@ void DigikamView::toggleShowBar(bool b)
     d->stackedview->thumbBarDock()->showThumbBar(b);
 }
 
-bool DigikamView::isThumbBarVisible()
-{
-    return d->stackedview->thumbBarDock()->isVisible();
-}
-
 void DigikamView::setRecurseAlbums(bool recursive)
 {
     d->iconView->imageAlbumModel()->setRecurseAlbums(recursive);
diff --git a/digikam/views/digikamview.h b/digikam/views/digikamview.h
index 669eac9..9229fa6 100644
--- a/digikam/views/digikamview.h
+++ b/digikam/views/digikamview.h
@@ -79,7 +79,6 @@ public:
     void hideSideBars();
     void setThumbSize(int size);
     void toggleShowBar(bool);
-    bool isThumbBarVisible();
     void setRecurseAlbums(bool recursive);
     void setRecurseTags(bool recursive);
     void imageTransform(RotationMatrix::TransformationAction transform);
diff --git a/utilities/importui/main/importui.cpp b/utilities/importui/main/importui.cpp
index f136998..03b53bb 100644
--- a/utilities/importui/main/importui.cpp
+++ b/utilities/importui/main/importui.cpp
@@ -2588,7 +2588,7 @@ void ImportUI::showThumbBar(bool visible)
 
 bool ImportUI::thumbbarVisibility() const
 {
-    return d->view->isThumbBarVisible();
+    return d->showBarAction->isChecked();
 }
 
 }  // namespace Digikam
diff --git a/utilities/importui/views/importview.cpp b/utilities/importui/views/importview.cpp
index fb30c48..a15728d 100644
--- a/utilities/importui/views/importview.cpp
+++ b/utilities/importui/views/importview.cpp
@@ -775,11 +775,6 @@ void ImportView::toggleShowBar(bool b)
     d->StackedView->thumbBarDock()->showThumbBar(b);
 }
 
-bool ImportView::isThumbBarVisible()
-{
-    return d->StackedView->thumbBarDock()->isVisible();
-}
-
 void ImportView::scrollTo(const QString& folder, const QString& file)
 {
     CamItemInfo info  = camItemInfo(folder, file);
diff --git a/utilities/importui/views/importview.h b/utilities/importui/views/importview.h
index 4b48970..42e01b1 100644
--- a/utilities/importui/views/importview.h
+++ b/utilities/importui/views/importview.h
@@ -58,7 +58,6 @@ public:
     void hideSideBars();
     void setThumbSize(int size);
     void toggleShowBar(bool b);
-    bool isThumbBarVisible();
 
     void scrollTo(const QString& folder, const QString& file);
 
